dll/cpl/input/advanced_settings_page.c

```diff
@@ -3,13 +3,93 @@
 * FILE:            dll/cpl/input/advanced_settings_page.c
 * PURPOSE:         input.dll
 * PROGRAMMER:      Dmitry Chapyshev (dmitry@reactos.org)
+*                  Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
 */
 
 #include "input.h"
 
+BOOL g_bTextServiceIsOff = FALSE;
+
+BOOL LoadAdvancedSettings(HWND hwndDlg)
+{
+    HKEY hKey;
+    LONG error;
+    DWORD dwValue = FALSE, cbValue = sizeof(dwValue);
+
+    error = RegOpenKeyExW(HKEY_CURRENT_USER, L"Software\\Microsoft\\CTF", 0, KEY_READ, &hKey);
+    if (error != ERROR_SUCCESS)
+        return FALSE;
+
+    RegQueryValueExW(hKey, L"Disable Thread Input Manager", NULL, NULL,
+                     (LPBYTE)&dwValue, &cbValue);
```

Comment:
What if the registry value is a string? or a dword with a non-standard size? etc.
This needs: success check; value type check, and retrieved size check, before using the dwValue below.

```suggestion
    error = RegQueryValueExW(hKey,
                             L"Disable Thread Input Manager",
                             NULL,
                             &dwType,
                             (LPBYTE)&dwValue,
                             &cbValue);
    if ((error != ERROR_SUCCESS) || (dwType != REG_DWORD) || (cbValue != sizeof(dwValue)))
        dwValue = FALSE; // Default
```
(this would require a `DWORD dwType;` declaration of course)

Reply:
Fixed